sites/geohub/src/lib/stac/EarthSearchStac.ts

```diff
@@ -174,6 +174,57 @@ export default class EarthSearchStac implements StacTemplate {
 		return feature;
 	};
 
+	public generateProductFeature = async (
+		item: StacItemFeature,
+		productId: string
+	): Promise<DatasetFeature> => {
+		const itemUrl = item.links.find((l) => l.rel === 'self').href;
+		const providers: Tag[] = this.stacCollection.providers?.map((p) => {
+			return { key: 'provider', value: p.name };
+		});
+		const feature: DatasetFeature = {
+			type: 'Feature',
+			geometry: {
+				type: 'Polygon',
+				coordinates: [
+					[
+						[item.bbox[0], item.bbox[1]],
+						[item.bbox[0], item.bbox[3]],
+						[item.bbox[2], item.bbox[1]],
+						[item.bbox[2], item.bbox[3]],
+						[item.bbox[0], item.bbox[1]]
+					]
+				]
+			},
+			properties: {
+				id: generateHashKey(itemUrl),
+				name: `${this.stacCollection.title} - ${productId}`,
+				description: this.stacCollection.description,
+				license: this.stacCollection.license,
+				// collection_id: this.collection,
+				// product: productId,
+				url: itemUrl,
+				is_raster: true,
+				access_level: AccessLevel.PUBLIC,
+				tags: [
+					{ key: 'type', value: 'stac' },
+					{ key: 'stacApiType', value: 'api' },
+					{ key: 'stacType', value: 'cog' },
+					{ key: 'stac', value: this.stacId },
+					{ key: 'collection', value: this.collection },
+					{ key: 'item', value: item.id },
+					{ key: 'product', value: productId },
```

Comment:
Looks like except adding `{ key: 'product', value: productId }` tag, all other code in this new function is same with `generateDataSetFeature`. Can you try to use `generateDataSetFeature` to add product tag instead of creating new one?